framework/doc/content/source/csg/CSGSurface.md

```diff
@@ -0,0 +1,36 @@
+# CSGSurface
+
+This is an abstract class for any surface definition for [constructive solid geometry (CSG)](syntax/CSG/index.md) representations.
+The [source/csg/CSGBase.md] framework already contains definitions for some basic surfaces (planes, spheres, and cylinders), but developers of modules or MOOSE-based application can define additional custom surfaces using this abstract class.
+General information about implementing [!ac](CSG)-related methods can be found in [source/csg/CSGBase.md].
+
+## Defining a New Surface Type
+
+Any arbitrary or custom surface can be defined by inheriting from `CSGSurface`.
+At a minimum, the surface name, surface type, and boundary type, must be set, and two virtual methods that need to be defined.
+Additional information required by the constructor will depend on the specific surface to be defined.
+
+### Setting Coefficients
+
+Each surface should be defined by some equation, which therefore requires the definition of coefficients or other parameters.
+These coefficient values should be set and be returned through the `getCoefficients` method which will return a map of the coefficient strings to their values.
+These coefficient values are returned in the [!ac](JSON) file that is produced.
+For example, a plane is defined by the equation $ax + by + cz = d$ and so this method would return a map of the values for `a`, `b`, `c`, and `d`.
```

Comment:
Are there any requirements or suggestions we want to impose on the name of the coefficients, i.e. that they have to be single characters? I don't know what the conventions are for these coefficients

Reply:
I thought about this but then I couldn't think of a reason in which we would need to impose that because we are just using the string as the key for the JSON file and nothing more. As long as the coefficient "name" is a string, that is all that should matter. I am thinking there might be some super complex surface types in which you would need a longer coeff name. Even for spheres and cylinders we use `x0`, `y0` etc which are already 2 characters.